src/components/request/RequestForm.tsx

```diff
@@ -194,20 +215,76 @@ const RequestForm: FC = () => {
     }
   }, []);
 
+  // Temp. This throws if you try to read certain values too quickly.
+  // TODO: Add proper loader.
+  useEffect(() => {
```

Comment:
if you do your queries in the useOracleReader, you wont need to wait here

Reply:
Removed